src/ApplicationConfig.php

```diff
@@ -23,50 +20,27 @@
     private $separator;
 
     /**
-     * @api
-     *
-     * @param array  $patterns
+     * @param array  $config
      * @param string $separator
-     *
-     * @return ApplicationConfig
      */
-    public static function fromFiles(array $patterns, $separator = '.')
+    public function __construct(array $config, $separator = '.')
     {
-        $locator = new FileLocator();
-        $files   = $locator->locate($patterns);
-
-        if (empty($files)) {
-            throw NoMatchingFilesException::fromPatterns($patterns);
-        }
-
-        $factory = new ReaderFactory([
-            '.json' => 'TomPHP\ConfigServiceProvider\FileReader\JSONFileReader',
-            '.php'  => 'TomPHP\ConfigServiceProvider\FileReader\PHPFileReader',
-        ]);
-
-        $configs = array_map(
-            function ($filename) use ($factory) {
-                $reader = $factory->create($filename);
-
-                return $reader->read($filename);
-            },
-            $files
-        );
-
-        $config = call_user_func_array('array_replace_recursive', $configs);
+        $this->config    = $config;
+        $this->separator = $separator;
+    }
 
-        return new self($config, $separator);
+    public function merge(array $config)
+    {
+        $this->config = array_replace_recursive($this->config, $config);
     }
 
     /**
-     * @api
-     *
-     * @param array  $config
      * @param string $separator
+     *
+     * @return void
      */
-    public function __construct(array $config, $separator = '.')
+    public function setSeparator($separator)
     {
-        $this->config    = $config;
         $this->separator = $separator;
```

Comment:
@tomphp 

Same as https://github.com/tomphp/config-service-provider/pull/37/files?short_path=04c6e90#r77787578, what do you think?